modules/shells/posix.nix

```diff
@@ -0,0 +1,67 @@
+# options and implementation for all POSIX shells
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+with lib; {
+  # common options for POSIX compatible shells
+  options = {
+    shell = {
+      aliases = mkOption {
+        example = {
+          ll = "ls -l";
+        };
+        description = lib.mdDoc ''
+          An attribute set that maps aliases (the top level attribute names in
+          this option) to command strings or directly to packages.
+          Aliases mapped to `null` are ignored.
+        '';
+        type = with types; attrsOf (nullOr (either str path));
+      };
+      hook = mkOption {
+        default = "";
+        description = lib.mdDoc ''
+          Shell script code called during environment activation.
+          This code is assumed to be shell-independent, which means you should
+          stick to pure sh without sh word split.
+        '';
+        type = types.lines;
+      };
+    };
+  };
+
+  config = let
+    stringAliases = concatStringsSep "\n" (
+      mapAttrsFlatten (k: v: "alias ${k}=${escapeShellArg v}")
+      (filterAttrs (k: v: v != null) config.shell.aliases)
+    );
+
+    exportedEnvVars = let
+      # make foo = "bar" -> foo = ["bar"]
+      allValuesLists =
+        mapAttrs (n: toList) config.variables;
```

Comment:
Needs an escapeShellArg or quoting or some disambiguating of when the string should undergo shell evaluation or be a direct string.

Reply:
Am I missing that somewhere in nixpkgs? https://github.com/NixOS/nixpkgs/blob/45b92369d6fafcf9e462789e98fbc735f23b5f64/nixos/modules/config/shells-environment.nix#L26